code-challenges.test.js

```diff
@@ -20,10 +22,41 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+// describe("arrayShuffle", () => {
+//     it("Takes in an array, removes the first item from the array and shuffles the remaining items.", () => {
+//         expect(arrayShuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+//         expect(arrayShuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+//     })
+// })
+
 // b) Create the function that makes the test pass.
 
+const arrayShuffle = (array) => {
+    array.shift()
+    let newArr = []
+    let arrIndex = []
+
+    while (arrIndex.length < array.length) {
+    let randomIndex = Math.floor(Math.random() * array.length)
+    
+    if (!arrIndex.includes(randomIndex)) {
+        arrIndex.push(randomIndex)
+    }
+}
+    for (let i = 0; i < arrIndex.length; i++) {
+        newArr.push(array[arrIndex[i]])
+    }
+    return newArr
+}
+console.log(arrayShuffle(colors1))
+console.log(arrayShuffle(colors2))
+
 // Pseudo code:
 
+// Input: an array
+// Output: a new array that has removed the first element from the original array and shuffled the remaining elements
+// Create a function that takes in an array as the parameter. Use .shift() method to remove the first element from the array. Use a while loop to generate an array of random indices. Use a for loop to push the remaining elements from the original array into a new array according to random indices. 
+
```

Comment:
Good job, Alex.  This is an interesting, out of the box way to approach this solution.  
Your pseudocode should be bullet point style rather than paragraph (just a note) - good explanation of the steps.